Allow custom config through script metadata

disenchant opened the pull request:
Unfortunately git2kandji hardcoded some configuration values for scripts like the name of the library item becoming the script name, execution frequency always being set to once and the script never being shown in Kandji self service.

This PR adds the capability to git2kandji to include configuration metadata into audit scripts which are then used to configure the library item accordingly.

As an example, scripts can now look like this:

```bash
#!/usr/bin/env bash

# git2kandji-config: name=Custom git2kandji Script Name
# git2kandji-config: active=false
# git2kandji-config: execution_frequency=every_15_min

echo "Audit: git2kandji"
```

PS: And I made the container image run as non-root user as a bonus 😉

Commit:
improve Dockerfile to run as non-root

Commit:
improved entrypoint

Commit:
script config through metadata

moojomoore commented on git2kandji.py:
```diff
@@ -555,15 +599,23 @@ def sync_kandji_scripts(local_scripts, kandji_scripts, dryrun=False):
         remediation_script = scripts.get('remediation')
 
+        # Parse metadata from the audit script (or remediation if audit is missing)
+        metadata_script = audit_script or remediation_script
```
I don't think it makes sense to parse metadata from the `remediation_script` as it's not possible to have a remediation script without an audit script. We also have a check just below this to make sure there isn't a remediation script without an audit script.

moojomoore replied:
```suggestion
        metadata_script = audit_script
```

disenchant replied:
I even thought about that when putting all together but somehow I left it there 🙈  Great catch and I fully agree with you that it doesn't make sense for the remediation scripts.

moojomoore committed:
Update git2kandji.py

Remove unnecessary script_name values with the latest custom config metadata changes

Commit:
add download option and metadata for profiles

moojomoore commented on Dockerfile:
This is looking great, thanks for thinking about the security here!

moojomoore commented on git2kandji.py:
Overall, everything looks great and all my tests worked successfully.

moojomoore reviewed (changes requested):
This is looking really good, thanks for putting this together!
I only had 1 change that I wanted your feedback on before approving.

I also removed the `script_name` values that were no longer needed with these changes.

Thanks again!

moojomoore commented:
Thanks @disenchant - When I was initially working on this, I was trying to decide the best way to handle this scenario, and then never came back to it. Initially, I was thinking an extra json file, but this is even better since there won't be extra files lying around.

Thanks again!

disenchant committed:
Update git2kandji.py

Co-authored-by: Jordan Moore <moojo.moore@gmail.com>

Commit:
Ensure script and profile directories exist before downloading

disenchant commented:
@moojomoore sorry for the interference but while you've tested it I was at the same time pushing an update. The last changes now implement a very similar approach for inline metadata for profiles and I've implemented a download flag which helps to get things started

moojomoore commented:
@disenchant - No problem, I'll review the latest changes this afternoon. Thanks for these awesome additions!

Commit:
bug fix

moojomoore committed:
Update git2kandji.py

moojomoore committed:
Create CHANGELOG.md
